build/src/nbb/build.clj

```diff
@@ -0,0 +1,72 @@
+(ns nbb.build
+  "Provides bb tasks for building and releasing nbb"
+  (:require [babashka.fs :as fs]
+            [babashka.classpath :as classpath]
+            [babashka.tasks :refer [shell clojure]]
+            [clojure.string :as str]
+            [clojure.edn :as edn]))
+
+(defn- feature-files
+  []
+  (filter fs/exists?
+          (map (fn [d]
+                 (fs/file d "nbb_features.edn"))
+               (classpath/split-classpath (classpath/get-classpath)))))
+
+(defn- read-configs
+  [files]
+  (->> files
+       (mapcat (comp edn/read-string slurp str))))
+
+(defn- build-cmd [cmd nbb-dir]
+  (let [files (feature-files)
+        feature-configs (read-configs files)
+        ;; Each ./src/nbb_features.edn has a ./deps.edn
+        feature-dirs (map (comp fs/parent fs/parent) files)
+        cmd' (if (seq files)
+               (format "-Sdeps '%s' %s"
+                       {:deps
+                        (merge (into {}
+                                     (map (fn [dir]
+                                            [(symbol (str (fs/file-name dir) "/deps"))
+                                             {:local/root (str dir)}])
+                                          feature-dirs))
+                               {'nbb/deps {:local/root nbb-dir}})}
+                       cmd)
+               cmd)]
+    (when (seq feature-configs)
+      (println "Building features:" (str/join ", " (map :name feature-configs)) "..."))
+    (if (seq feature-configs)
+      (apply str cmd'
+        (map (fn [m] (format " --config-merge '%s'" (pr-str (:shadow-config m))))
+             feature-configs))
+      cmd')))
+
+(defn build
+  "Build nbb shadow builds using clojure cmd and commandline args. Features on
+  classpath are automatically added"
+  [cmd args]
+  (let [building-outside-nbb? (not (fs/exists? "shadow-cljs.edn"))
+        nbb-dir (when building-outside-nbb?
+                  (->> (classpath/get-classpath)
+                       classpath/split-classpath
+                       ;; Pull out nbb from local/root or git/url
+                       (some #(when (re-find #"(nbb/[0-9a-f]+|nbb)/src" %) %))
+                       fs/parent))]
+    (when building-outside-nbb?
+      (fs/copy (fs/file nbb-dir "shadow-cljs.edn") "shadow-cljs.edn"))
+    (apply clojure (build-cmd cmd (str nbb-dir)) args)
+    (when building-outside-nbb?
+      (fs/delete "shadow-cljs.edn"))))
+
+(defn release
```

Comment:
I exposed build and release as the two main bb tasks. build is useful so I can also run dev builds. release is useful so that the nbb-logseq build just works in case their our any updates later on